### PR TITLE
Futher v2 changes

### PR DESCRIPTION
**Pull request by fleetwoods**

### CODE REVIEW

File: src/main/scala/com/ovoenergy/comms/model/TriggeredV3.scala [hunk 1]
```diff
@@ -21,12 +21,10 @@ case class TriggeredV3(
...
-    @AvroDoc(
       "An ordered list of the preferred channels to deliver the given Comm. Note that under some circumstances a Comm might be sent over a non-preferred channel if a customer has missing contact details in their account") preferredChannels: Option[
       List[Channel]])
     extends LoggableEvent {
-  import MetadataV2._
-  override def loggableString: Option[String] = prettyPrint(this, fieldsToRemove = Seq("templateData", "deliverTo"))
-  override def mdcMap: Map[String, String]    = combineMDCS(metadata.mdcMap, deliverTo.mdcMap)
+
+  override def loggableString: Option[String] = prettyPrint(this, fieldsToRemove = Seq("templateData"))
+  override def mdcMap: Map[String, String]    = combineMDCS(metadata.mdcMap)
```
Comment (laurence-bird) [inline src/main/scala/com/ovoenergy/comms/model/TriggeredV3.scala]: i guess you don't need to combine it if this is just the metdata MDC map now

File: src/test/scala/com.ovoenergy.comms/model/LoggableEventSpec.scala [hunk 2]
```diff
@@ -9,40 +12,57 @@ import org.scalacheck._
...
   property("TriggeredV3") = forAllNoShrink { msg: TriggeredV3 =>
     val traceToken = msg.mdcMap.get("traceToken")
     val commName   = msg.mdcMap.get("commName")
     val version    = msg.mdcMap.get("commVersion")
+    val customerId = msg.mdcMap.get("customerId")
 
     val isObfuscated = {
       val loggableString = msg.loggableString
-      !loggableString.get.contains(msg.deliverTo) && !loggableString.get.contains(msg.templateData)
+      !loggableString.get.contains(msg.templateData)
```
Comment (cb372) [inline src/test/scala/com.ovoenergy.comms/model/LoggableEventSpec.scala]: Not related to your change, but this line doesn't make any sense. `loggableString` (a string) will never contain `msg.templateData`.

TODO fix this test in a separate PR

File: src/main/scala/GenerateSchemas.scala [hunk 3]
```diff
@@ -35,11 +35,14 @@ object GenerateSchemas extends App {
...
   writeToFile(AvroSchema[FailedV2], "FailedV2")
   writeToFile(AvroSchema[FailedCancellation], "FailedCancellation")
+  writeToFile(AvroSchema[FailedCancellationV2], "FailedCancellationV2")
   writeToFile(AvroSchema[LinkClicked], "LinkClicked")
   writeToFile(AvroSchema[LinkClickedV2], "LinkClickedV2")
   writeToFile(AvroSchema[OrchestrationStartedV2], "OrchestrationStartedV2")
   writeToFile(AvroSchema[IssuedForDelivery], "IssuedForDelivery")
   writeToFile(AvroSchema[IssuedForDeliveryV2], "IssuedForDeliveryV2")
+  writeToFile(AvroSchema[CancellationRequested], "CancellationRequested")
+  writeToFile(AvroSchema[CancellationRequestedV2], "CancellationRequestedV2")
```
Comment (cb372) [inline src/main/scala/GenerateSchemas.scala]: Luckily nobody is actually using this feature yet, so this breaking change is only inconvenient for us...

File: src/main/scala/com/ovoenergy/comms/model/GenericMetadataV2.scala [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+case class GenericMetadataV2(
+                              @AvroDoc("A timestamp denoting when the event was created. Value is milliseconds since Unix Epoch - UTC time.") createdAt: Instant,
+                              @AvroDoc("A globally unique ID for this Kafka message. A UUID is suitable.") eventId: String,
+                              @AvroDoc("""
+  An ID that can be used to link all the requests and Kafka messages in a given transaction.
+  If you already have a trace token from a previous event/request, you should copy it here.
+  If this is the very start of a transaction, you should generate a fresh trace token and put it here.
+  A UUID is suitable.
+           """) traceToken: String,
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/comms/model/GenericMetadataV2.scala]: As soon as you merge this PR, I'm making another PR to fix the formatting of all these docstrings so we can actually read our code!

**Pull request merged**